src/nn/activations.rs

```diff
@@ -55,9 +60,175 @@ where
+pub struct LeakyReLU<E: Dtype>(E);
+
+impl<E: Dtype> Default for LeakyReLU<E> {
+    fn default() -> Self {
+        Self(E::from_f32(0.05).unwrap())
+    }
+}
+
+impl<E: Dtype> ZeroSizedModule for LeakyReLU<E> {}
+impl<E: Dtype> NonMutableModule for LeakyReLU<E> {}
+
+impl<S: ConstShape, E: Dtype, D: Device<E>, T: Tape<E, D>> Module<Tensor<S, E, D, T>>
+    for LeakyReLU<E>
+where
+    Tensor<S, E, D, T>: TryPReLU<E, Output = Tensor<S, E, D, T>>,
+{
+    type Output = Tensor<S, E, D, T>;
+    type Error = D::Err;
+
+    fn try_forward(&self, input: Tensor<S, E, D, T>) -> Result<Self::Output, D::Err> {
+        Ok(leakyrelu(input, self.0))
+    }
+}
+
+pub mod builder {
+    #[derive(Debug, Copy, Clone, Eq, PartialEq)]
+    pub struct PReLU;
+}
+
+impl<E: Dtype, D: Device<E>> BuildOnDevice<D, E> for builder::PReLU
+where
+    PReLU<E, D>: BuildModule<D, E>,
+{
+    type Built = PReLU<E, D>;
+    fn try_build_on_device(device: &D) -> Result<Self::Built, <D>::Err> {
+        Self::Built::try_build(device)
+    }
+}
+
+/// Calls [prelu()] with learnable value.
+#[derive(Debug, Clone)]
+pub struct PReLU<E: Dtype, D: Device<E>> {
+    a: Tensor<(), E, D>,
+}
+
+impl<E: Dtype, D: Device<E>> Default for PReLU<E, D> {
+    fn default() -> Self {
+        let dev = D::default();
+        Self {
+            a: dev.tensor(E::from_f32(0.25).unwrap()),
+        }
+    }
+}
+
+impl<E: Dtype, D: Device<E>> From<E> for PReLU<E, D> {
+    fn from(value: E) -> Self {
+        let dev = D::default();
+        Self {
+            a: dev.tensor(value),
+        }
+    }
+}
+
+impl<E: Dtype, D: Device<E>> NonMutableModule for PReLU<E, D> {}
+
+impl<S: ConstShape, E: Dtype, D: Device<E>, T: Tape<E, D>> Module<Tensor<S, E, D, T>>
+    for PReLU<E, D>
+where
+    Tensor<S, E, D, T>: TryPReLU<Tensor<S, E, D, NoneTape>, Output = Tensor<S, E, D, T>>,
+{
+    type Output = Tensor<S, E, D, T>;
+    type Error = <Tensor<S, E, D, T> as HasErr>::Err;
+
+    fn try_forward(&self, input: Tensor<S, E, D, T>) -> Result<Self::Output, Self::Error> {
+        input.try_prelu(self.a.clone().broadcast())
+    }
+}
+
+impl<E: Dtype, D: Device<E>> TensorCollection<E, D> for PReLU<E, D> {
+    type To<E2: Dtype, D2: Device<E2>> = PReLU<E2, D2>;
+
+    fn iter_tensors<V: crate::prelude::ModuleVisitor<Self, E, D>>(
+        visitor: &mut V,
+    ) -> Result<Option<Self::To<V::E2, V::D2>>, V::Err> {
+        visitor.visit_fields(
+            Self::tensor("a", |p| &p.a, |p| &mut p.a, TensorOptions::reset_to_zeros()),
+            |a| PReLU { a },
+        )
+    }
+}
+
+/// Calls [prelu()] with learnable values along second dimension.
+#[derive(Debug, Clone)]
+pub struct PReLU1D<C: ConstDim, E: Dtype, D: Device<E>> {
+    a: Tensor<(C,), E, D>,
+}
+
+impl<C: ConstDim, E: Dtype, D: Device<E>> Default for PReLU1D<C, E, D> {
+    fn default() -> Self {
+        let dev = D::default();
+        Self {
+            a: dev.tensor(E::from_f32(0.25).unwrap()).broadcast(),
+        }
+    }
+}
```

Comment:
Let's remove this to make it consistent with the other nn modules (that you have to go through BuildModule)

Reply:
done